docs/_posts/2022-06-24-types-as-comments-proposal.md

```diff
@@ -0,0 +1,170 @@
+
+![State of JS 2020: Satisfaction vs Usage](/assets/2022-06_state-of-js-satisfaction-vs-usage.png)
+
+TODO note on type inference makes explicit static typing optional
+
+## Making type annotations part of JavaScript proper
+
+### Motivation
+
+Having established that static typing is generally useful for JavaScript development (and development in general), the question to address is this: why not just develop in TypeScript if we want to add static type checks to our JavaScript development process? 
+
+One big argument in favor of types as comments in JavaScript compared to TypeScript (or similar languages and tools) is that it gives us the choice of type checker we want to use. We could for example opt for [Flow](https://flow.org/) type checker instead of TypeScript's type checker if we preferred that (and if Flow started to support the types as comments syntax).
+
+Another important argument pro types as comments in JavaScript is that it lets us skip the inevitable JavaScript transpilation step. Compare that to developing in TypeScript, where the usual workflow includes running our code through `tsc` (the TypeScript compiler) before executing it with `node` or in the browser. Dropping the transpilation step may significantly improve the speed of our development loop and moreover may require less set-up (such as the `tsconfig.json` file).
+
+![Comparison of development workflow: type as comments and TypeScript](/assets/2022-06_types-as-comments-comparison.png)
+
+> :information_source: A more elaborate discussion can be found here in [this article on the Microsoft dev blog](https://devblogs.microsoft.com/typescript/a-proposal-for-type-syntax-in-javascript/). (Picture above is also taken from the blog.)
+> 
+
+### Some examples
+
+The [proposal for types as comments][tc39-types-as-comments] will probably not confront anyone who has worked with either TypeScript or Flow with mind-bending concepts. Rather, the proposed syntax for types (and the associated features of the type system) very closely resembles that of TypeScript and Flow.
+
+A **type annotation** is written in the form `x: T` where `x` is a variable name and `T` is a type name. We may use type annotations in variable declarations or in function parameters.
+
+```typescript
+// a variable that stores a string
+let x: string; 
+// a function that takes two string parameters and returns a boolean value
+function equals(s: string, t: string): boolean { /* … */ }
+```
+
+Types may be **declared** using the `interface` keyword to introduce a new type or using the `type` keyword to declare aliases.
+
+```typescript
+// An object type with a properties coordinate and radius.
+interface Circle {
+  coordinate: Point;
+  radius: number
+}
+// A shorter alias for boolean
+type bool = boolean;
+```
+
+Classes also implicitly define types.
+
+```typescript
+// Implicitly declares a type Point
+class Point {
+  x: number;
+  y: number
+}
+```
```

Comment:
As it's great to look into upcoming features, the whole syntax looks like regular TypeScript code. A reader might want to know what's the difference between TypeScript and the JS proposal.

Reply:
There mostly are none. The whole thing is heavily inspired by TS and Flow. I can add a short paragraph which states that explicitly though. (thumbsup)